packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1521,6 +1566,61 @@ export class TransactionController extends BaseController<
   ): boolean {
     return remoteGasUsed !== localGasUsed;
   }
+
+  private requestApproval(txMeta: TransactionMeta) {
+    const id = this.getApprovalId(txMeta);
+    const { origin } = txMeta;
+    const type = ApprovalType.Transaction;
+
+    this.messagingSystem
+      .call(
+        'ApprovalController:addRequest',
+        {
+          id,
+          origin: origin || ORIGIN_METAMASK,
+          type,
+          requestData: {
+            id: txMeta.id,
```

Comment:
We shouldn't need to duplicate the `id` and `origin` here as they are already available as part of the approval request.

Reply:
Indeed, removing the duplicates and simplifying the request data